### PR TITLE
[Test] Make tests to compare compressed and non-compressed obs

### PR DESCRIPTION
**Pull request by vincentpierre**
### Proposed change(s)

 - Make `Utilities.TextureToTensorProxy` public and renamed it to `writer.WriteTexture`
 - GridSensor can now be compressed or uncompressed
 - Added 2 scenes and an integration test to make sure compressed and uncompressed obs are the same

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [x] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

**Comment by vincentpierre** [comment]
Wow, thank @Phantomb !

File: .yamato/gym-interface-test.yml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 test_editors:
-  - version: 2019.4
```
Comment (chriselion) [inline .yamato/gym-interface-test.yml:2]: Did you intend to make all the version changes part of the PR?

Comment (vincentpierre) [reply .yamato/gym-interface-test.yml:2]: Yes. These tests were actually not running at all since the dependency (test-build) was not run for this version

Comment (chriselion) [reply .yamato/gym-interface-test.yml:2]: Looks like we need to run the standalone build in 2019.4 not 2019.3; then you can revert these.

File: Project/Assets/ML-Agents/TestScenes/TestCompressedTexture/TestTextureSensor.cs [hunk 2]
```diff
@@ -0,0 +1,72 @@
+using UnityEngine;
+using Unity.MLAgents.Sensors;
+using Unity.MLAgents;
+
+public class TestTextureSensor : ISensor
+{
+    Texture2D m_Texture;
+    bool m_Grayscale;
```
Comment (chriselion) [inline Project/Assets/ML-Agents/TestScenes/TestCompressedTexture/TestTextureSensor.cs:8]: Unused?

**Comment by chriselion** [comment]
(Sorry, can't comment on the file moves)
Why do we need to move all the SharedAsset files?

The texture you're using is symmetric on the horizontal axis, so it wouldn't detect some of the bugs we're trying to catch. Can you pick something that is asymmetric (and also non-square)?

**Review by dongruoping — COMMENTED** [review]
Test looks good to me.

Might not be the focus of this PR but this reveals an issue to me that the stacking sensor initializes the compressed obs buffer with empty png and the uncompressed buffer with zeros, and empty png is not exactly the same as zero array. So if you do env.step() less than number of stacks it could probably fail.

File: ml-agents/tests/yamato/scripts/run_compressed_sensor.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    """
+    env = UnityEnvironment(
+        file_name=env_name, no_graphics=True, additional_args=["-logFile", "-"]
+    )
+
+    try:
+        # Reset the environment
+        env.reset()
+
+        env.step()
```
Comment (chriselion) [inline ml-agents/tests/yamato/scripts/run_compressed_sensor.py:22]: Why multiple steps? Maybe set the DecisionRequester period to 1?

File: .yamato/gym-interface-test.yml (hunk 1 above)
Comment (vincentpierre) [reply .yamato/gym-interface-test.yml:2]: I tried that and it said the bokken image was not found.

File: ml-agents/tests/yamato/scripts/run_compressed_sensor.py (hunk 3 above)
Comment (vincentpierre) [reply ml-agents/tests/yamato/scripts/run_compressed_sensor.py:22]: The decision requester is already set to 1. I wanted the stacking to be complete

**Comment by vincentpierre** [comment]
> Why do we need to move all the SharedAsset files?
It bothered me that it was part of Examples. I can revert that no problem if you think it is unnecessary.

**Comment by chriselion** [comment]
>  I can revert that no problem if you think it is unnecessary.
No strong feelings

**Comment by vincentpierre** [comment]
> Might not be the focus of this PR but this reveals an issue to me that the stacking sensor initializes the compressed obs buffer with empty png and the uncompressed buffer with zeros, and empty png is not exactly the same as zero array. So if you do env.step() less than number of stacks it could probably fail.

You are spot on. This is a bug and the test is failing because of it. Any suggestions on how to fix it ?

**Comment by dongruoping** [comment]
> You are spot on. This is a bug and the test is failing because of it. Any suggestions on how to fix it ?
To make them the same, initialize the compressed buffer with PNGs that equals to all zeros after decompression. That means we have to create a texture equals to all zeros, which I'm not sure how to do it. Currently it's using a "empty" texture and somehow that's not zeros.

File: com.unity.ml-agents/Runtime/Sensors/StackingSensor.cs [hunk 4]
```diff
@@ -221,6 +221,14 @@ internal byte[] CreateEmptyPNG()
             int height = m_WrappedSensor.GetObservationShape()[0];
             int width = m_WrappedSensor.GetObservationShape()[1];
             var texture2D = new Texture2D(width, height, TextureFormat.RGB24, false);
+            for (int h = 0; h < texture2D.height; h++)
+            {
+                for (int w = 0; w < texture2D.width; w++)
+                {
+                    texture2D.SetPixel(w, h, Color.black);
```
Comment (chriselion) [inline com.unity.ml-agents/Runtime/Sensors/StackingSensor.cs:228]: I believe it's bad practice to call SetPixel repeatedly like this.

Maybe something like this?
https://answers.unity.com/questions/635383/how-can-i-change-all-of-a-textures-pixels-to-one-c.html

File: .yamato/training-int-tests.yml [hunk 5]
```diff
@@ -1,7 +1,5 @@
 test_editors:
   - version: 2018.4
```
Comment (chriselion) [inline .yamato/training-int-tests.yml:2]: Can you add 2020.1 to standalone-build-test.yml, add it back here, and add 2019.3 here?

**Review by chriselion — APPROVED** [review]
One comment about the test versions, otherwise looks good.

Can you update the PR description about TextureToTensorProxy too?

**Pull request merged**